src/content/docs/tips/script-tag-dynamic-imports.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: Dynamic imports in module scripts
+description: How dynamic imports offer granular control over fetch and execution order
+astroRange: ^4.0.0
+---
+
+Astro bundles and processes `<script>` tags into [`type="module"`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) for us automatically, which makes scripts execute similiar to [`defer`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer) by default. This is great for a balance of performance and time to interactivity/hydration. 
+
+With [Interaction to Next Paint](https://web.dev/blog/inp-cwv-launch) (INP) officially replacing [First Input Delay](https://web.dev/articles/fid) (FID) it might seem absurd to suggest replacing a static import inside a module script to a dynamic one, but if done correctly it can be useful sometimes.
+
+## Granular control
+
+### Static Imports
+
+Static imports ensure that your code is hydrated and ready to run as soon as possible, without affecting [First Contentful Paint](https://web.dev/articles/fcp) (FCP) or [Largest Contentful Paint](https://web.dev/articles/lcp) (LCP). This is important for INP because it ensures that any interactive elements are ready to react to interaction immediately.
+
+### Dynamic Imports
+
+Dynamic imports provide fine-grained control over when your module is imported (downloaded). Since static imports will all be fetched, parsed, and evaluated at the beginning of a module script's execution, importing two expensive modules in the same module script tag can cause delays in the execution of code that only depends on the first module. Dynamic imports allow you to either import a module only when it is needed, or dictate the order of imports allowing for manual optimization of network requests.
+
+### Controlling Network Requests
+
+Dynamic imports also allow you to control the sequence of network requests. Instead of fetching all modules in parallel, which is the default behavior when using multiple static imports or separate module scripts, dynamic imports enable you to ensure that one module is downloaded first before initiating the fetch for the next module. This can be particularly useful when you have modules that are critical for immediate user interaction and others that are less critical or if you are supporting users with very low bandwidth.
+
+## Example
```

Comment:
I think the examples below can be improved by 1:
- Using real modules and using them (ie. must be basic to avoid distracting, you can use EC line markers)
- Explain why using 1 example or another is better for those modules
Overall being more practical I guess

Reply:
Any ideas for real world modules? I used confetti for the example repo but wasn't creative enough to come up with real real world examples lol

Reply:
i mean confetti is better than "..."!